GKE_docker_image_tool/maxtext.Dockerfile

```diff
@@ -0,0 +1,34 @@
+# Use Python 3.10 as the base image
+FROM python:3.10-slim
+
+# Install system dependencies and Git
+RUN apt-get update && apt-get install -y curl gnupg git
+
+# Add the Google Cloud SDK package repository
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+
+# Install the Google Cloud SDK
+RUN apt-get update && apt-get install -y google-cloud-sdk
+
+# Set the default Python version to 3.10
+RUN update-alternatives --install /usr/bin/python3 python3 /usr/local/bin/python3.10 1
+
+# Set environment variables for Google Cloud SDK and Python 3.10
+ENV PATH="/usr/local/google-cloud-sdk/bin:/usr/local/bin/python3.10:${PATH}"
+
+ARG MODE
+ENV ENV_MODE=$MODE
+
+ARG JAX_VERSION
+ENV ENV_JAX_VERSION=$JAX_VERSION
+
+# Set the working directory in the container
+WORKDIR /app
+
+RUN git clone https://github.com/google/maxtext.git
+
+RUN echo "Running command: bash setup.sh MODE=$ENV_MODE JAX_VERSION=$ENV_JAX_VERSION"
+RUN cd maxtext && bash setup.sh MODE=${ENV_MODE} JAX_VERSION=${ENV_JAX_VERSION}
```

Comment:
This workflow is a good start but doesn't seem to address the typical user flow.

Imagine the user's flow is (1) make a change to MaxText locally, (2) test it locally either (2a) in a docker image or (2b) in on the raw VM and then (3) upload a docker image to GKE.

This script assumes that you're grabbing code from MaxText off of Git (so it doesn't work with local changes) and the rebuilds will be slow because you assume we want to reinstall dependencies each time (even though 95+ of changes don't change the dependencies).

Instead, we want:
* use the local version of MaxText (so the user can easily make changes). (Notice that if the user wants the one on github they can just `git clone` it!
* support gracefully either rebuilding with dependencies or just copying over the local MaxText. (usually there isn't a change to the dependencies so there is no reason to setup.sh) I believe Docker handles this via layering where the dependencies would be a different layer.
* [More a question] How would the user develop locally in their docker container? This could either be running a train.py or doing an XAOT and it might be invoked as part of building or it might be interactive.
* [Totally a nit] Can the MaxText directory be the root directory in the docker image? It is super annoying that scripts need to `cd maxtext` in the Docker image but not outside. I wonder if docker can be set to have a specific default working directory?

Reply:
1) 2) 4) done